repl/src/main/scala/ammonite/repl/Storage.scala

```diff
@@ -0,0 +1,53 @@
+package ammonite.repl
+
+import acyclic.file
+import java.io.{File, FileInputStream, IOException, FileWriter}
+import org.yaml.snakeyaml.Yaml
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.JavaConversions.asScalaBuffer
+
+object Storage{
+  val dir = new java.io.File(System.getProperty("user.home") + "/.ammonite")    
+  if(dir.exists){
+    if(!dir.isDirectory){
+      dir.delete()
+      dir.mkdir()
+    }
+  } else {
+    dir.mkdir()
+  }
+
+  lazy val yaml = new Yaml
+
+  def loadHistory: History = {
+    try{
+      val res = new History
+      val list = yaml.synchronized{
+        yaml.load(new FileInputStream(dir + "/history"))
+      }
+      list match {
+        case a: java.util.List[String] => res ++= a
+        case _ =>
+      }
+      res
+    } catch {
+      case e: IOException => new History
+    }
+  }
+
+  def saveHistory(h: History): Unit = {
```

Comment:
How long does dumping the whole histo ry every time take? Picking some arbitrary metric like 1000 lines of history 100 characters each? 10000 lines? If it turns out to be slow we should think about this more (e.g. append to the file instead of re-writing the whole thing). In either case, the rough benchmarks should live in a comment in the source code